### ISSUE TITLE
SA1009 is reported instead of SA1022

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#256]
In the following code, SA1009 is reported for the `)` token. In this situation, SA1009 should be suppressed and `SA1022` should be reported for the `+` token.

``` csharp
var x = (int) +3;
```
